frontend/server/src/main/java/org/pytorch/serve/http/ManagementRequestHandler.java

```diff
@@ -270,13 +270,14 @@ private void handleUnregisterModel(
         } else if (httpResponseStatus == HttpResponseStatus.BAD_REQUEST) {
             throw new ModelVersionNotFoundException(
                     String.format(
-                            "Model version: %s not found for model: %s", modelVersion, modelName));
+                            "Model version: %s does not exist for model: %s",
+                            modelVersion, modelName));
         } else if (httpResponseStatus == HttpResponseStatus.INTERNAL_SERVER_ERROR) {
             throw new InternalServerException("Interrupted while cleaning resources: " + modelName);
         } else if (httpResponseStatus == HttpResponseStatus.REQUEST_TIMEOUT) {
             throw new RequestTimeoutException("Timed out while cleaning resources: " + modelName);
         } else if (httpResponseStatus == HttpResponseStatus.FORBIDDEN) {
-            throw new InternalServerException(
+            throw new InvalidModelVersionException(
```

Comment:
Should this be 'throw ModelVersionNotFoundException'? Also, let's keep only one class.

Reply:
InvalidModelVersionException for case where user tries to unregister the default version.